server/src/main/java/org/elasticsearch/index/mapper/ParametrizedFieldMapper.java

```diff
@@ -145,6 +146,7 @@ protected final void doXContentBody(XContentBuilder builder, boolean includeDefa
         private boolean acceptsNull = false;
         private Consumer<T> validator = null;
         private Serializer<T> serializer = XContentBuilder::field;
+        private Predicate<Boolean> serializerPredicate = id -> true;
```

Comment:
Should this just be a `BooleanSupplier`? If `includeDefaults` is false would you ever include it?

Reply:
It feels weird to make it a `Predicate` because it isn't really a predicate *on* the boolean, it is a predicate on the state of the field mapper.

Reply:
You're quite right, have updated.